doc/decisions/library_directory_structure.md

```diff
@@ -2,21 +2,75 @@
 
 ## Problem
 
-Currently src/libs/elektra contains the source of several libraries. Each of the libraries should have its own folder.
+Currently `src/libs/elektra` contains the source of several libraries. Each of the libraries should have its own folder.
 Furthermore, we need to allow rust source to exist in parallel.
 
 ## Constraints
 
+- It must be possible to set up include-paths via CMake so that `#include "header.h"` works the same in the build directory and when installed.
+
 ## Assumptions
 
 ## Considered Alternatives
 
 ## Decision
 
+We use this directory tree (YAML syntax for highlighting):
+
+```yaml
+doc:
+examples:
+scripts:
+src:
+  include:
+   # headers that don't belong to a specific library
+  libs:
+    core-c: C implementation of libelektra-core.so
+    core-rust: Rust implementation of libelektra-core.so
+    opts: libelektra-opts.so has only a single implementation, so no suffix
+    mount-c: mounting library written in C
```

Comment:
I would only give -c prefixes if an alternative implementation exists. We plan to remove the C++ tools lib.

```suggestion
```